File: Python/733. Flood Fill.py
class Solution:
    def floodFill(self, image, sr, sc, newColor):
        """
        :type image: List[List[int]]
        :type sr: int
        :type sc: int
        :type newColor: int
        :rtype: List[List[int]]
        """
        dx = [0, 1, -1, 0]
        dy = [1, 0, 0, -1]
        n, m = len(image), len(image[0])
        maple = dict()
        q = list()
        color = image[sr][sc]
        image[sr][sc] = newColor
        maple[(sr, sc)] = 1
        q.append((sr, sc))
        while len(q) > 0:
            node = q.pop(0)
            for i in range(0,4):
                tx = node[0] + dx[i]
                ty = node[1] + dy[i]
                if tx < 0 or ty < 0:
                    continue
                if tx >= n or ty >= m:
                    continue
                if image[tx][ty] == color and maple.get((tx, ty)) is None:
                    maple[(tx, ty)] = 1
                    image[tx][ty] = newColor
                    q.append((tx, ty))
        return image

File: C++/733. Flood Fill.cpp
class Solution {
public:
    vector<vector<int>> floodFill(vector<vector<int>>& image, int sr, int sc, int newColor) {
        int vis[100][100] = {0};
        int dx[] = {0,1,-1,0};
        int dy[] = {1,0,0,-1};
        int color = image[sr][sc];
        int n = image.size(),m = image[0].size();
        queue<pair<int,int> >q;
        q.push({sr,sc});
        vis[sr][sc] = 1;
        image[sr][sc] = newColor;
        while(!q.empty())
        {
            pair<int,int> now = q.front();
            q.pop();
            for(int i=0;i<4;i++)
            {
                int tx = now.first + dx[i];
                int ty = now.second + dy[i];
                if(tx<0 || ty<0 || tx>=n || ty>=m)
                    continue;
                if(image[tx][ty]==color && !vis[tx][ty])
                {
                    vis[tx][ty] = 1;
                    image[tx][ty] = newColor;
                    q.push({tx,ty});
                }
            }
        }
        return image;
    }
};